### PR TITLE
Prometheus: Azure scopes from Grafana Azure SDK

### PR DESCRIPTION
**Pull request by kostrse**
**What is this feature?**

New release of [grafana/grafana-azure-sdk-go](https://github.com/grafana/grafana-azure-sdk-go) introduced cloud-specific settings which can be used instead of keeping the settings inside of the datasource itself.

**Why do we need this feature?**

Refactoring to use the new API of Grafana Azure SDK.

**Which issue(s) does this PR fix?**:

Related #61124

**Special notes for your reviewer:**

No changes to the end user.

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by itsmylife — APPROVED** [review]
LGTM

One question: Could you please provide one or two unit tests that covers this new situation?

**Comment by itsmylife** [comment]
Also tagging @grafana/partner-datasources for their review too.

**Review by aangelisc — APPROVED** [review]
👍 for me but as @itsmylife requested, some unit tests may be useful here

File: pkg/tsdb/prometheus/azureauth/azure.go [hunk 1]
```diff
@@ -82,11 +74,25 @@ func getPrometheusScopes(settings *azsettings.AzureSettings, credentials azcrede
 		return nil, err
 	}
 
+	cloudSettings, err := settings.GetCloud(azureCloud)
```
Comment (aangelisc) [inline pkg/tsdb/prometheus/azureauth/azure.go:77]: Shouldn't this be:
```suggestion
	cloudSettings, err := azureSettings.GetCloud(azureCloud)
```

**Comment by kostrse** [comment]
> 👍 for me but as @itsmylife requested, some unit tests may be useful here
Make cosmetic changes for readability and added tests.

File: pkg/tsdb/prometheus/azureauth/azure_test.go [hunk 2]
```diff
@@ -115,3 +116,31 @@ func TestConfigureAzureAuthentication(t *testing.T) {
...
+func TestGetPrometheusScopes(t *testing.T) {
+	cfg := &setting.Cfg{
+		Azure: &azsettings.AzureSettings{
+			Cloud: azsettings.AzureUSGovernment,
+		},
+	}
+
+	t.Run("should return scopes for cloud from settings with MSI credentials", func(t *testing.T) {
+		credentials := &azcredentials.AzureManagedIdentityCredentials{}
+		scopes, err := getPrometheusScopes(cfg.Azure, credentials)
```
Comment (itsmylife) [inline pkg/tsdb/prometheus/azureauth/azure_test.go:129]:
```suggestion
		scopes, err := getPrometheusScopes(cfgAzure, credentials)
```

File: pkg/tsdb/prometheus/azureauth/azure_test.go [hunk 3]
```diff
@@ -115,3 +116,31 @@ func TestConfigureAzureAuthentication(t *testing.T) {
...
+		require.NoError(t, err)
+
+		assert.NotNil(t, scopes)
+		assert.Len(t, scopes, 1)
+		assert.Equal(t, "https://prometheus.monitor.azure.us/.default", scopes[0])
+	})
+
+	t.Run("should return scopes for cloud from client secret credentials", func(t *testing.T) {
+		credentials := &azcredentials.AzureClientSecretCredentials{AzureCloud: azsettings.AzureChina}
+		scopes, err := getPrometheusScopes(cfg.Azure, credentials)
```
Comment (itsmylife) [inline pkg/tsdb/prometheus/azureauth/azure_test.go:139]:
```suggestion
		scopes, err := getPrometheusScopes(cfgAzure, credentials)
```

**Comment by itsmylife** [comment]
There is this PR I made and it was merged yesterday https://github.com/grafana/grafana/pull/81813
Could you please merge `main` into your branch? There is a build issue in CI I think that'll resolve it.

**Comment by kostrse** [comment]
> There is this PR I made and it was merged yesterday #81813 Could you please merge `main` into your branch? There is a build issue in CI I think that'll resolve it.

Updated tests.

**Comment by itsmylife** [comment]
@kostrse Thank you very much.

**Pull request merged**